ml_models/bpe_model.py

```diff
@@ -0,0 +1,71 @@
+from functools import wraps
+import numpy as np
+from text_utils.indexer import *
+from text_utils.utils import create_logger
+from config import logger_path
+from bpemb import BPEmb
+
+logger = create_logger(__name__, logger_path['bpe'])
+
+FEATURE_SIZE = 300
+
+def singleton(cls):
+    instance = None
+
+    @wraps(cls)
+    def inner(*args, **kwargs):
+        nonlocal instance
+        if instance is None:
+            instance = cls(*args, **kwargs)
+        return instance
+
+    return inner
+
+class BPEEmbedder:
+    def __init__(self):
+        self.model = self.build_model()
+        self.success_count = 0
+        self.error_count = 0
+
+    @singleton
+    def build_model(self):
+        model = BPEmb(lang="multi", vs=1000000, dim=FEATURE_SIZE)
+        return model
+
+    def sentence_embedding(self, text):
+        try:
+            mean_emb = self.model.embed(text).mean(axis=0)
+            self.success_count += 1
+            return mean_emb / np.linalg.norm(mean_emb)
+        except:
+            self.error_count += 1
+        return np.zeros(FEATURE_SIZE)
+
+def check_indexer():
+    test_queries = ['Есть ли аналоги pandas (ну или не аналоги а тоже либы для работы с данными) для работы с данными',
```

Comment:
Duplicated code, it can be replaced to new file `examples.py`

Reply:
To be fixed in separate patch.